File: examples/collection/collection.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')


def statements():
    # doc-start
    statements = statement.Collection().allow_ec2_instance_delete_by_owner()
    # doc-end
    return statements


all = statements()
out(all)
deploy(all)

File: examples/collection/collection.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Collection } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Collection().allowEc2InstanceDeleteByOwner()
    // doc-end
  );
}
const s = getStatement();
out(s);
deploy(s);
